src/Libraries/Microsoft.Extensions.Diagnostics.ResourceMonitoring/Windows/WindowsCounters.cs

```diff
@@ -10,203 +11,289 @@ internal sealed class WindowsCounters
 {
     public WindowsCounters(IMeterFactory meterFactory, TcpTableInfo tcpTableInfo)
     {
+        const string NetworkStateKey = "system.network.state";
+        const string InstrumentName = "process.network.connections";
+
 #pragma warning disable CA2000 // Dispose objects before losing scope
         // We don't dispose the meter because IMeterFactory handles that
         // An issue on analyzer side: https://github.com/dotnet/roslyn-analyzers/issues/6912
         // Related documentation: https://github.com/dotnet/docs/pull/37170
         var meter = meterFactory.Create("Microsoft.Extensions.Diagnostics.ResourceMonitoring");
 #pragma warning restore CA2000 // Dispose objects before losing scope
 
-        _ = meter.CreateObservableGauge(
-            "ipv4_tcp_connection_closed_count",
+        var tcpTag = new KeyValuePair<string, object?>("network.transport", "tcp");
+
+        // These are covered in https://github.com/open-telemetry/semantic-conventions/blob/main/docs/rpc/rpc-metrics.md#attributes:
+        var tcpVersionFourTag = new KeyValuePair<string, object?>("network.type", "ipv4");
+        var tcpVersionSixTag = new KeyValuePair<string, object?>("network.type", "ipv6");
+
+        // These metrics are aligned with
+        // https://github.com/open-telemetry/semantic-conventions/blob/main/docs/system/system-metrics.md#metric-systemnetworkconnections
+
+        // IPv4:
+        _ = meter.CreateObservableUpDownCounter(
+            InstrumentName,
             () =>
             {
                 var snapshot = tcpTableInfo.GetIPv4CachingSnapshot();
                 return snapshot.ClosedCount;
-            });
+            },
+            unit: "{connection}",
```

Comment:
nit: InstrumentName is a constant. Might as well do the same with unit to reduce duplication and to be consistent.

Reply:
Not applicable anymore since we have just one instrument